aggregator/src/aggregator/aggregation_job_creator.rs

```diff
@@ -347,6 +380,164 @@ impl<C: Clock + 'static> AggregationJobCreator<C> {
             .await?)
     }
 
+    #[tracing::instrument(skip(self), err)]
+    async fn create_aggregation_jobs_for_fixed_size_task_no_param<
+        const L: usize,
+        A: vdaf::Aggregator<L, AggregationParam = ()>,
+    >(
+        self: Arc<Self>,
+        task: Arc<Task>,
+        max_batch_size: usize,
+    ) -> anyhow::Result<()>
+    where
+        for<'a> &'a A::AggregateShare: Into<Vec<u8>>,
+        A::PrepareMessage: Send + Sync,
+        A::PrepareState: Send + Sync + Encode,
+        A::OutputShare: Send + Sync,
+        for<'a> &'a A::OutputShare: Into<Vec<u8>>,
+    {
+        Ok(self
+            .datastore
+            .run_tx(|tx| {
+                let (this, task) = (Arc::clone(&self), Arc::clone(&task));
+                Box::pin(async move {
+                    let min_batch_size = usize::try_from(task.min_batch_size())?;
+
+                    // Find unaggregated client reports & existing unfilled batches.
+                    let (mut unaggregated_report_ids, outstanding_batches) = try_join!(
+                        tx.get_unaggregated_client_report_ids_for_task(task.id()),
+                        tx.get_outstanding_batches_for_task(task.id())
+                    )?;
+
+                    // First attempt to allocate unaggregated reports to existing unfilled batches,
+                    // then generate new batches as necessary.
+                    let mut batch_iter = outstanding_batches
+                        .into_iter()
+                        .map(|outstanding_batch| (false, outstanding_batch))
+                        .chain(iter::repeat_with(|| {
+                            (true, OutstandingBatch::new(*task.id(), random(), 0, 0))
+                        }));
+
+                    // Main loop: repeatedly consume some of the unaggregated report IDs to generate
+                    // an aggregation job, assigning it to an existing batch which has need of
+                    // reports, or a new batch if no existing batch needs reports.
+                    let mut aggregation_jobs = Vec::<AggregationJob<L, FixedSize, A>>::new();
+                    let mut report_aggregations = Vec::<ReportAggregation<L, A>>::new();
+                    let mut new_batches = Vec::new();
+                    let (mut is_batch_new, mut batch) = batch_iter.next().unwrap();
+                    let mut max_size = batch.max_size();
```

Comment:
It's a bit confusing resolving the notion of an `OustandingBatch`'s `max_size` against the task's `max_size`. IIUC `OutstandingBatch.max_size` is the biggest batch that could be constructed based on the reports currently assigned to the batch, but in fact the batch could still get bigger if we end up assigning more reports, right? Maybe `OutstandingBatch.potential_size` or (perhaps too wordy) `OutstandingBatch.current_potential_size` is more clear?

Reply:
Correct: `min_size` & `max_size` track the potential size of the batch, depending on how currently-incomplete aggregations resolve. Specifically, `min_size` tracks how many reports are included in the batch if all reports currently undergoing aggregation eventually fail (i.e. it is the number of report aggregations in state `FINISHED`). `max_size` tracks how many reports are included in the batch if all reports currently undergoing aggregation eventually succeed (i.e. it is the number of report aggregations in any non-terminal-failure state).

Let's chat about the names:
* I like not including `current` -- the reasoning here is that every model is the "current" state of the underlying data [at some point in time, I suppose], so that word is basically implied for every field, or at least every field that is effectively mutable over time. That is, by the same logic we would justify calling the `state` field contained by many data models `current_state` instead.
* `potential_size` has a couple of issues that I think need to be resolved to get to a good naming scheme:
  * An fair (IMO) interpretation is that every size between `min_size` and `max_size` (inclusive) is a "potential" size of the batch. We could call this `max_potential_size` but then we're kind of back to the original problem, I guess.
  * If we interpret the max size as the "potential" size, we still need a good name for the minimum size.
  * I considered just making a single field `potential_size: Range<usize>`, tracking the entire range of possible batch sizes. Unfortunately, Rust's `Range` type is exclusive, which would make the logic using this field more complex/error-prone (i.e. off-by-one errors) than otherwise necessary. Still, I can go ahead w/ this strategy if you think it wise.

IMO, `min_size` & `max_size` are OK since they are attached to an `OutstandingBatch` object. I did go ahead and update the variable naming in `aggregation_job_creator.rs` to attempt to ensure clarity between a task's max_batch_size & a batch's size.

Reply:
There is std::ops::RangeInclusive, FWIW

Reply:
[I realized you're traveling, so I'm going to submit this PR without waiting to hear back on this issue, but poke this if you have an idea & I'll address in a followup PR.]

Reply:
I don't know how I missed that, I was looking specifically for it. Thanks, `OutstandingBatch` now has a single `size` method.